Packages/Core/Runtime/Events/AtomEvent.cs

```diff
@@ -104,63 +108,68 @@ public override void UnregisterAll()
         }
 
         /// <summary>
-        /// Register a Listener that in turn trigger all its associated handlers when the Event triggers.
+        /// Register a Listener that in turn trigger all its associated handlers when the Event is invoked.
         /// </summary>
         /// <param name="listener">The Listener to register.</param>
         public void RegisterListener(IAtomListener<T> listener, bool replayEventsBuffer = true)
         {
             _onEvent += listener.OnEventRaised;
             if (replayEventsBuffer)
-            {
                 ReplayBufferToSubscriber(listener.OnEventRaised);
-            }
         }
 
         /// <summary>
-        /// Unregister a listener that was registered using the `RegisterListener` method.
+        /// Unregister a listener that was registered using the <see cref="RegisterListener"/> method.
         /// </summary>
         /// <param name="listener">The Listener to unregister.</param>
         public void UnregisterListener(IAtomListener<T> listener)
         {
             _onEvent -= listener.OnEventRaised;
         }
 
-        #region Observable
         /// <summary>
-        /// Turn the Event into an `IObservable&lt;T&gt;`. Makes Events compatible with for example UniRx.
+        /// Turn the Event into an <see cref="IObservable{T}"/>. Makes Events compatible with for example UniRx.
+        /// </summary>
+        /// <returns>The Event as an <see cref="IObservable{T}"/>.</returns>
+        public IObservable<T> Observe() => new ObservableEvent<T>(Register, Unregister);
+
+        /// <summary>
+        /// Adds <paramref name="item"/> to the replay buffer.
         /// </summary>
-        /// <returns>The Event as an `IObservable&lt;T&gt;`.</returns>
-        public IObservable<T> Observe()
+        /// <param name="item">The item to add.</param>
+        protected void AddToReplayBuffer(T item)
         {
-            return new ObservableEvent<T>(Register, Unregister);
+            if (_replayBufferSize <= 0) return;
+            TrimReplayBuffer();
+            _replayBuffer.Enqueue(item);
         }
-        #endregion // Observable
 
-        protected void AddToReplayBuffer(T item)
+        /// <summary>
+        /// Trim the replay buffer by removing excess items.
+        /// </summary>
+        private void TrimReplayBuffer()
         {
-            if (_replayBufferSize > 0)
-            {
-                while (_replayBuffer.Count >= _replayBufferSize) { _replayBuffer.Dequeue(); }
-                _replayBuffer.Enqueue(item);
-            }
+            while (_replayBuffer.Count >= _replayBufferSize)
+                _replayBuffer.Dequeue();
         }
```

Comment:
this change does nothing for readability as i now have to scan 10+ LOC more, it does nothing for DRY (since the method is only used at this one place).
furthermore: when introducing new methods, one must carry about not putting them between those two, since it would even more harm readability as i know have to scroll.

A maybe better change to this would be creating a RingBuffer class, which cares about keeping the size constant. that would be a favor in terms of separation of concerns.

It may introduce a (probably negligible) performance impact - hopefully the compiler will inline this. (a local function would always be inlined)

Reply:
Sure, I'll roll this out.

I was considering adding another class similar to your RingBuffer suggestion, but that felt a little over engineered. I'm not actually not super keen on the fact that the event is handling it's own replay buffer at all. Feels like that should behaviour should be "pluggable" per Event, but that's for another day :-).